cirq/ops/eigen_gate.py

```diff
@@ -319,6 +319,25 @@ def _resolve_parameters_(self: TSelf, param_resolver) -> TSelf:
         return self._with_exponent(
                 exponent=param_resolver.value_of(self._exponent))
 
+    def _equal_up_to_global_phase_(self, other, atol):
+        if not isinstance(other, EigenGate):
+            return NotImplemented
+
+        exponents = (self._canonical_exponent, other._canonical_exponent)
+        exponents_is_parameterized = tuple(
+            protocols.is_parameterized(e) for e in exponents)
+        if (all(exponents_is_parameterized) and exponents[0] != exponents[1]):
+            return False
+        if (any(exponents_is_parameterized) or
+                not np.isclose(exponents[0], exponents[1], atol=atol)):
+            return False
+
+        self_without_exp_or_phase = self._with_exponent(0)
```

Comment:
This part confuses me.  1) why does it do a creation of two new objects? 2) what does setting _with_exponent(0) followed by setting global_shift to 0 leave to be different?

Reply:
.